src/components/pages/add-reservation/ReserveForm.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable */
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { addNewReserve } from '../../../redux/features/reservations/mine';
+import { getRooms } from '../../../redux/roomsSlice';
+
+function ReserveForm() {
+    const dispatch = useDispatch();
+    const { data, loading, error } = useSelector((state) => state.rooms);
+
+    useEffect(() => {
+        dispatch(getRooms());
+    }, [dispatch]);
```

Comment:
- Kindly, complete all the expected behaviors when fetching available rooms bearing in mind that users can only reserve rooms that aren't currently reserved.